src/wallet/rpcwallet.cpp

```diff
@@ -77,6 +77,20 @@ static bool ParseIncludeWatchonly(const UniValue& include_watchonly, const CWall
 }
 
 
+static CWallet* GetReadyWallet(std::shared_ptr<CWallet> const wallet) {
+    CWallet* const pwallet = wallet.get();
+
+    // Make sure the results are valid at least up to the most recent block
+    // the user could have gotten from another RPC command prior to now
+    pwallet->BlockUntilSyncedToCurrentChain();
+
+    // Ensure that the mempool has been synced at least once on startup
+    if (!pwallet->mempoolSynced()) pwallet->syncMempool();
```

Comment:
d37756f380b27d259a7818078ddf02caa0a4578a

Just call `pwallet->syncMempool()` and early return there - no need to expose mempool synced flag.